rajawali/src/main/java/org/rajawali3d/view/TextureView.kt

```diff
@@ -920,11 +923,24 @@ open class TextureView @JvmOverloads constructor(
                 guardedRun()
             } catch (e: InterruptedException) {
                 // fall thru and exit normally
+            } catch (e: IllegalStateException) {
+                Log.e("RajawaliGLThread", e.message)
+                showToast(e.message)
+            } catch (e: Exception) {
+                Log.e("RajawaliGLThread", e.message)
             } finally {
                 glThreadManager.threadExiting(this)
             }
         }
 
+        fun showToast(text: String?) {
+            if (Looper.myLooper() == Looper.getMainLooper())
```

Comment:
Not much value in checking the looper and you don't need to worry about passing in the looper to the handler. This can be reduced to simply

```kotlin
Handler().post { Toast.makeTest(context, text, Toast.LENGTH_LONG).show() }
```